File: codegen/results/v002/m3-max-48gb/lmstudio/qwen_qwen3-coder-30b_nothink/20260828_113901/python/002_word_frequency/solution.py
import re
from collections import Counter

# Read the file
with open('input/text.txt', 'r') as file:
    content = file.read()

# Convert to lowercase and extract words (only letters)
words = re.findall(r'[a-zA-Z]+', content.lower())

# Count frequency of each word
word_counts = Counter(words)

# Sort by count descending, then alphabetically ascending
sorted_words = sorted(word_counts.items(), key=lambda x: (-x[1], x[0]))

# Output results
for word, count in sorted_words:
    print(f"{word}: {count}")

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen_qwen3-coder-30b_nothink/20260828_113901/typescript/002_word_frequency/solution.ts
import fs from 'fs';
import path from 'path';

const filePath = path.join('input', 'text.txt');

// Read the file content
const content = fs.readFileSync(filePath, 'utf-8');

// Split into words and process them
const words = content
  .split(/\s+/)
  .map(word => 
    // Remove punctuation and convert to lowercase
    word.toLowerCase().replace(/[^a-z]/g, '')
  )
  .filter(word => word.length > 0); // Filter out empty strings

// Count word frequencies
const wordCount: { [key: string]: number } = {};
for (const word of words) {
  wordCount[word] = (wordCount[word] || 0) + 1;
}

// Sort by count descending, then alphabetically ascending
const sortedEntries = Object.entries(wordCount).sort(([wordA, countA], [wordB, countB]) => {
  if (countB !== countA) {
    return countB - countA; // Descending count
  }
  return wordA.localeCompare(wordB); // Ascending alphabetical
});

// Output results
for (const [word, count] of sortedEntries) {
  console.log(`${word}: ${count}`);
}